doc/guides/writing_tests.md

```diff
@@ -0,0 +1,166 @@
+# How to write a Node.js test
```

Comment:
Would "How to write a test for the Node.js project" be better because I thought this is for how to write a test case in Node.js community, not the Node.js project in my first sight on this title. :-)